### PR TITLE
docs(messaging): add warning for `react-native-splash-screen` users

### PR DESCRIPTION
**Pull request by jasperkuperus**
### Description

Using React Native Firebase together with `react-native-splash-screen` might cause some issues. Especially for `messaging().getInitialNotification()` and `messaging().onNotificationOpenedApp`. This PR adds a small comment to the documentation of these two methods to warn people about known issues.

### Related issues

Related issue where this was discussed: https://github.com/invertase/react-native-firebase/issues/3469#issuecomment-660121376

### Release Summary

N/A

### Checklist

- I read the [Contributor Guide](../CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
  - [x] Yes
- ~My change supports the following platforms;~
  - ~[] `Android`~
  - ~[ ] `iOS`~
- ~My change includes tests;~
  - ~[ ] `e2e` tests added or updated in `packages/\*\*/e2e`~
  - ~[ ] `jest` tests added or updated in `packages/\*\*/__tests__`~
- ~[ ] I have updated TypeScript types that are affected by my change.~
- This is a breaking change;
  - [ ] Yes
  - [x] No

### Test Plan

N/A -> Documentation only

:fire:

### CODE REVIEW

File: packages/messaging/lib/index.d.ts [hunk 1]
```diff
@@ -549,6 +549,9 @@ export namespace FirebaseMessagingTypes {
      *
      * See `onNotificationOpenedApp` to subscribe to when the notification is opened when the app
      * is in a background state.
+     *
+     * Beware of this [issue](https://github.com/invertase/react-native-firebase/issues/3469#issuecomment-660121376)
+     * when you're using `react-native-splash-screen`.
```
Comment (mikehardy) [inline packages/messaging/lib/index.d.ts:554]:
```suggestion
     * Beware of this [issue](https://github.com/invertase/react-native-firebase/issues/3469#issuecomment-660121376) when integrating with splash screen modules. If you are using
     * `react-native-splash-screen` we strongly recommend you migrate to `react-native-bootsplash`
     * which is actively maintained and avoids these issues
```

File: packages/messaging/lib/index.d.ts [hunk 2]
```diff
@@ -653,6 +656,9 @@ export namespace FirebaseMessagingTypes {
      * See `getInitialNotification` to see how to watch for when a notification opens the app from a
      * quit state.
      *
+     * Beware of this [issue](https://github.com/invertase/react-native-firebase/issues/3469#issuecomment-660121376)
+     * when you're using `react-native-splash-screen`.
```
Comment (mikehardy) [inline packages/messaging/lib/index.d.ts:660]:
```suggestion
     * Beware of this [issue](https://github.com/invertase/react-native-firebase/issues/3469#issuecomment-660121376) when integrating with splash screen modules. If you are using
     * `react-native-splash-screen` we strongly recommend you migrate to `react-native-bootsplash`
     * which is actively maintained and avoids these issues
```

**Comment by jasperkuperus** [comment]
@mikehardy Accepted your proposed changes 1 on 1.

**Comment by Ehesp** [comment]
Thanks for this. It'll require the `gen:reference` script to be re-run but one of us can sort this (assuming you didn't clone locally).

**Comment by jasperkuperus** [comment]
> Thanks for this. It'll require the `gen:reference` script to be re-run but one of us can sort this (assuming you didn't clone locally).

Thanks. Indeed I didn't clone locally.

**Comment by JDMathew** [comment]
This comment saved me... thanks @jasperkuperus

**Pull request merged**